pkg/kubelet/kubelet_pods.go

```diff
@@ -924,24 +924,13 @@ func countRunningContainerStatus(status v1.PodStatus) int {
 	return runningContainers
 }
 
-// PodCouldHaveRunningContainers returns true if the pod with the given UID could still have running
-// containers. This returns false if the pod has not yet been started or the pod is unknown.
-func (kl *Kubelet) PodCouldHaveRunningContainers(pod *v1.Pod) bool {
-	if kl.podWorkers.CouldHaveRunningContainers(pod.UID) {
-		return true
-	}
-
-	// Check if pod might need to unprepare resources before termination
-	// NOTE: This is a temporary solution. This call is here to avoid changing
-	// status manager and its tests.
-	// TODO: extend PodDeletionSafetyProvider interface and implement it
-	// in a separate Kubelet method.
+// ShouldDelayTransitioningPodToTerminal returns true if transitioning pod to terminal state should be delayed.
+func (kl *Kubelet) ShouldDelayTransitioningPodToTerminal(pod *v1.Pod) bool {
```

Comment:
I don't see any significant changes in this PR except for renamings which risk unnecessary conflicts for other PRs just before the deadline. 

Reading the left note
`
// NOTE: This is a temporary solution. This call is here to avoid changing
// status manager and its tests.
`
it seems there was something more to be done.

Reply:
I really do not like adding new methods to the pod state machine right before code freeze over other changes. 
 I will review this today but this is code that is highly sensitive and has been the source of numerous breaks over the years.

Reply:
@mimowo

> I don't see any significant changes in this PR except for renamings which risk unnecessary conflicts for other PRs just before the deadline.

The PR did what was mentioned in the TODO - it added new method to the PodDeletionSafetyProvider in its first edition.  It was considered dangerous by reviewers and we decided to extend and rename `PodCouldHaveRunningContainers` instead. This change is in the second commit and I can drop it any time if needed. Please look if the first commit looks more reasonable, thanks.